gateway/ln-gateway/src/lnd.rs

```diff
@@ -286,12 +289,21 @@ impl ILnRpcClient for GatewayLndClient {
         Ok(GetRouteHintsResponse { route_hints })
     }
 
-    async fn pay(&self, invoice: PayInvoiceRequest) -> crate::Result<PayInvoiceResponse> {
+    async fn pay(&self, request: PayInvoiceRequest) -> crate::Result<PayInvoiceResponse> {
+        let PayInvoiceRequest {
+            invoice,
+            max_fee_percent,
+            ..
+        } = request;
+
         let mut client = self.client.clone();
         let send_response = client
             .lightning()
             .send_payment_sync(SendRequest {
-                payment_request: invoice.invoice.to_string(),
+                payment_request: invoice.to_string(),
+                fee_limit: Some(FeeLimit {
+                    limit: Some(Limit::Percent(max_fee_percent.round() as i64)),
```

Comment:
- need to validate there's no data loss and safety in casting `max_fee_percent`
- need to validate `max_fee_percent` is respected by LND pay

Reply:
How would this *not* lead to loss of information. Can we just change this limit to an absolute value? I don't know what made me use % in the first pace.

Reply:
LND gRPC as proxied by `tonic_lnd` demands an i64 which seems odd because why would a scenario want to set negative percentages as fee limit? I think this should be a u64 similar to what CLN pay api expects.

Reply:
It might make sense to cast and then explicitly check if the value is < 0 (and throw error if it is), just to be sure

Reply:
Ack. What do you think of pre-computing a fixed value based on the % and fee amount, and rounding that?

Reply:
I'm baffled that the API seems to actually represent a percentage as an integer without any multiplier (e.g. making it a parts per million value instead). We really need to switch to absolute values, integer % granularity is horrible.

https://github.com/lightningnetwork/lnd/blob/0d8d0fd261402ac7b00deec4b01d5f6183302858/lnrpc/lightning.proto#L774C1-L775

Reply:
Migrated to absolute values

Reply:
Still, LNDs use of unsigned values might trip us in case we ever need to set very large fee limits `fee_limit_msat > i64::MAX`, but I think this should be a super rare case. Should it happen, this PR proposes we fall back to [LNDs default fee limits as specified here](https://github.com/lightningnetwork/lnd/blob/0d8d0fd261402ac7b00deec4b01d5f6183302858/lnrpc/lightning.proto#L835:L838)

Reply:
`i64::MAX` msat = 92M BTC if I didn't make a mistake, so likely not a problem :laughing: